### DISCUSSION TITLE
Frigate recommended servers and AVX

### DISCUSSION THREAD

**Discussion by gatorreina** [blakeblackshear/frigate#3320] [General]
I am currently using the Frigate recommended Minisforum GK41 to run Frigate. Evidently its J4125 chipset does not have AVX technology which prevents me from being able to run Compreface and use it in conjunction with Frigate. https://github.com/exadel-inc/CompreFace/issues/577 . I have noticed that the other top two Frigate recommendations, the Odessey X86 and the Beelink GK55 also have the J4125 chipset. Does anyone know of a work-around for this or of a dual NIC mini computer that works well for both Frigate and Compreface?

**Comment by kbrouder** [comment]
Doesnt answer your question, but I am using this addon https://github.com/jakowenko/double-take. I am using it with Deepstack, but it supports compreface. I am running deepstack on a jetson nano separate from my frigate instance, so no hardware conflicts...Possibly you can go that approach?

**Reply by gatorreina** [reply]
Hello kbrouder. Thanks for your input. I am have double-take running as well. I have not been able to get deepstack working due to another issue which is probably a cause of my lack of experience with Docker Compose. I will continue to try to find a way to get deepstack working, however, I would like to try to able to get Compreface working with Frigate and the only thing that appears to be impeding me from doing so is this chipset issue.

**Comment by blakeblackshear** [comment]
This can't be true. Frigate also requires AVX since it uses tensorflow lite under the hood. There is a long standing issue here on how to make it possible to run Frigate on non-AVX CPUs: https://github.com/blakeblackshear/frigate-hass-addons/pull/12

**Reply by blakeblackshear** [reply]
Also, I personally use machines with J4125.

**Reply by gatorreina** [reply]
Hello Blake, Thank you very much for the reply and for all your work on Frigate. Are you running Compreface as a Docker container on any of those J4125 machines? I have Frigate working without issue as docker container via docker compose. My problem is that I cannot get Compreface to work via docker compose. The only thing I see in the Compreface container logs is: !!! no internal routing support, rebuild with pcre support !!!

**Reply by NickM-27** [reply]
Which compreface version are you using? There are three different types:

(from compreface)
```
  FaceNet(default), (compreface:1.0.0)  - For general purposes. Support CPU without AVX2 
 
  Arcface-R100, (compreface:1.0.0-arcface-r100) - The most accurate model, but the most slowest 
 
  MobileNet, (compreface:1.0.0-mobilenet) - The fastest model among CPU only models 
```

**Reply by gatorreina** [reply]
Hi Nick, Thanks for the reply. I followed this: https://github.com/exadel-inc/CompreFace/blob/master/custom-builds/FaceNet/docker-compose.yml and am using compreface-core.

**Reply by NickM-27** [reply]
Have you made an issue with compreface? I imagine a different image would solve the issue or maybe it's something else

**Reply by gatorreina** [reply]
Hi Nick, By an issue with compreface do you mean another container with an image of just plain compreface? If so can you point me in the direction of docs that show how to set up such a image/container in docker compose?

**Reply by NickM-27** [reply]
I mean a GitHub issue asking for support, they an tell for sure what the issue is and how to fix.

**Reply by gatorreina** [reply]
Ok, thanks will do.

**Comment by felipecrs** [comment]
I also have a J4125 and I was never able to make CompreFace work with it, unfortunately. I tried years ago, and stopped here:

https://github.com/exadel-inc/CompreFace/issues/583#issuecomment-997106979

It should be possible with a knowledgeable person working on a new variant of the CompreFace container that applies these changes.

@blakeblackshear, you mentioned that it should support AVX, but according to [Intel ARK](https://www.intel.com/content/www/us/en/products/sku/197305/intel-celeron-processor-j4125-4m-cache-up-to-2-70-ghz/specifications.html), it does not.

That said, I still suggest mentioning in the suggested hardware documentation that CompreFace and DeepStack containers does not support running on such machines, at least to date.

**Comment by guttermonk** [comment]
Anyone have any luck finding a dual NIC mini PC with AVX support?

**Reply by felipecrs** [reply]
Beelink EQ12.

**Reply by guttermonk** [reply]
That mini PC has the N100 processor and on the [Intel ark site](https://ark.intel.com/content/www/us/en/ark/products/231803/intel-processor-n100-6m-cache-up-to-3-40-ghz.html) it lists AVX2 under the instruction set extensions. Is AVX2 backwards compatible with AVX, and were you able to get CompreFace to work on it?

**Reply by felipecrs** [reply]
I hope so, I bought one for my brother but it didn't arrive yet. Someday I'll test CompreFace with it.

**Reply by Cold-Lemonade** [reply]
@felipecrs I have a Beelink EQ12. If you plan on using coral USBs, you may need to play around in the BIOS to boost the power to the usb ports. I didn't explore that route. Instead, I use a powered usb hub to get my Beelink EQ12 to work with them. This was not an issue with my Minisforum GK41.

Also, if you're going to run Ubuntu on it, my understanding (and I could easily be wrong) is that you'll want to install 23.04 because earlier versions do not offer the best power optimization for the Intel N100.

And one more thing about the Beelink EQ12, if you plan on using the usb-c port, you may need to get a cable with a 10mm connector because the plastic case is so thick it prevents standard usb-c cables from fully connecting to the port. Overall, it is a nice little computer, but it seems to have this design flaw with respect to the usb-c port.

**Reply by felipecrs** [reply]
Thank you! BTW:

1. I don't plan to use any Coral, I plan to use OpenVINO instead.
2. I plan to run Home Assistant OS, which since v10 has the necessary drivers for Intel N100

**Reply by Cold-Lemonade** [reply]
Please let us know how well OpenVino works on your Beelink EQ12. I'm curious what kind of detection speeds you get.

**Reply by guttermonk** [reply]
I bought an EQ12, migrated to HAOS, and can confirm it works with Frigate and Coral. Just finished setting it up without any issues, besides looking at old forum configs. This project took the importance of RTFM to the next level. Good times. Also, curious about OpenVino.

**Reply by felipecrs** [reply]
@andrew-lemon since you mentioned Ubuntu: It seems that Linux 6.1 is now available on Ubuntu 22.04 LTS through HWE.

**Comment by gatorreina** [comment]
I have done two Frigate->Doubletake->Compreface setups with this Beelink SEi8. Has the AVX support but only one NIC. Sent from my iPhoneOn Mar 28, 2023, at 11:26 AM, Felipe Santos ***@***.***> wrote:﻿
I also have a J4125 and I was never able to make CompreFace work with it, unfortunately. I tried years ago, and stopped here:
exadel-inc/CompreFace#583 (comment)
It should be possible with a knowledgeable person working on a new variant of the CompreFace container that applies these differences.
@blakeblackshear, you mentioned that it should support AVX, but according to Intel ARK, it does not.
That said, I still suggest mentioning in the suggested hardware documentation that CompreFace and DeepStack containers does not support running on such machines, at least to date.
